NEWS.md

```diff
@@ -14,6 +14,10 @@ New language features
   * Named tuples, with the syntax `(a=1, b=2)`. These behave very similarly to tuples,
     except components can also be accessed by name using dot syntax `t.a` ([#22194]).
 
+  * Keyword argument containers (`kw` in `f(; kw...)`) are now named tuples. Dictionary
+    functions like `haskey` and indexing can be used on them, and name-value pairs can be
+    iterated using `pairs(kw)` ([#4916]).
+
```

Comment:
I wonder if it'd be worth mentioning that keyword arguments can no longer be repeated; I know that wasn't changed explicitly in this PR, but seems like a good place to mention it anyway.